libr/core/cmd_seek.c

```diff
@@ -273,30 +273,41 @@ static void seek_to_register(RCore *core, const char *input, bool is_silent) {
 }
 
 static int cmd_seek_opcode_backward(RCore *core, int n) {
-	int val = 0;
+	int i, val = 0;
 	// N previous instructions
 	ut64 addr = core->offset;
 	int ret = 0;
 	int numinstr = n * -1;
 	if (r_core_prevop_addr (core, core->offset, numinstr, &addr)) {
 		ret = core->offset - addr;
+		r_core_seek (core, addr, true);
 	} else {
 #if 0
 		// core_asm_bwdis_len is buggy as hell we should kill it. seems like prevop_addr
 		// works as expected, because is the one used from visual
 		ret = r_core_asm_bwdis_len (core, &instr_len, &addr, numinstr);
 #endif
-		ut64 prev_addr = prevop_addr (core, core->offset);
-		if (prev_addr <= core->offset) {
-			RAsmOp op;
+		const int mininstrsize = r_anal_archinfo (core->anal, R_ANAL_ARCHINFO_MIN_OP_SIZE);
+		ut64 addr = core->offset;
+		for (i = 0; i < numinstr; i++) {
+			ut64 prev_addr = prevop_addr (core, addr);
+			if (prev_addr == UT64_MAX) {
+				prev_addr = addr - mininstrsize;
+			}
+			if (prev_addr >= core->offset) {
+				break;
+			}
+			RAsmOp op = {0};
 			r_core_seek (core, prev_addr, 1);
-			r_asm_disassemble (core->assembler, &op,
-					core->block, 32);
+			r_asm_disassemble (core->assembler, &op, core->block, 32);
+			if (op.size < mininstrsize) {
```

Comment:
can this ever happen? if it does, I'd say there's a bug in `r_anal_archinfo (core->anal, R_ANAL_ARCHINFO_MIN_OP_SIZE);`

Reply:
when disassembling unaligned instructions the unaligned is shown as size 3, which is < 4, therefor it can happen.

Reply:
i tested on arm-init in r2r and even on unaligned instructions, it always returned 4. Anyway, let's keep it for now, it's not even that important in this case to adjust the value because `val` is just used as the value for core->num